### PR TITLE
Fixes #7: UFGroup required fields bug

### LINKED ISSUE #7: Fields are sometimes marked as required when they shouldn't be
The fieldmetadata API:

```php
$result = civicrm_api3('Fieldmetadata', 'get', array(
  'entity' => "UFGroup",
  'entity_params' => array('id' => 35),
));
```

... is a little inconsistent with regard to a field's `required` property. While the property's value always comes back as a string, it's "1" for each field that is specified as required in the profile, "" for fields where the default "not required" setting was left untouched, or "0" where it was explicitly declared as not required.

The display layer correctly handles "" fields – these are displayed as not required – but "0" fields are treated as though they were "1" fields.

Proposed solution:
* The metadata API should be explicit and consistent. Let's make sure "not required" fields always have a `required` value of "0."
* Update the display layer. If required is "1," display the field as required; any other value is treated as "not required."

### PR DESCRIPTION
**Pull request by universalhandle**

### CODE REVIEW

File: CRM/Fieldmetadata/Normalizer.php [hunk 1]
```diff
@@ -97,6 +97,22 @@ function getEmptyOption() {
...
+   * When output from normalizers is predictable and consistent, clients have
+   * less type juggling to do.
+   *
+   * This method might be out of place here. It may be more appropriate to have
+   * a Field class than to address this level of detail here.
+   *
+   * @param mixed $value
+   * @return string "1" or "0"
+   */
+  function stringifyBooleanValue($value) {
```
Comment (caracolmzd) [inline CRM/Fieldmetadata/Normalizer.php]: I would prefer "normalizePhpBoolean"

**Comment by universalhandle** [comment]
Merged after incorporating feedback from MZD.

File: CRM/Fieldmetadata/Normalizer/UFGroup.php [hunk 2]
```diff
@@ -30,7 +29,7 @@ function normalizeData(&$data, $params) {
       $field["label"] = $fieldData['title'];
       $field["name"] = $fieldData['name'];
       $field["order"] = $fieldOrder;
-      $field["required"] = $fieldData['is_required'];
+      $field["required"] = $this->stringifyBooleanValue($fieldData['is_required']);
```
Comment (caracolmzd) [inline CRM/Fieldmetadata/Normalizer/UFGroup.php:32]: Looks like you missed renaming this call.

Comment (universalhandle) [reply CRM/Fieldmetadata/Normalizer/UFGroup.php:32]: This was already fixed.

**Pull request merged**